elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/framework/SQLUnitTest.java

```diff
@@ -535,4 +590,60 @@ protected void testQueryExecution(Query query) {
+                        + "            PARSEDATETIME(FORMATDATETIME(`com_yahoo_elide_datastores_aggregation_example_PlayerStats`.`recordedDate`, 'yyyy-MM'), 'yyyy-MM') AS `recordedDate_155839778` \n"
+                        + "        FROM \n"
+                        + "            `playerStats` AS `com_yahoo_elide_datastores_aggregation_example_PlayerStats` \n"
+                        + "            LEFT OUTER JOIN \n"
+                        + "            `countries` AS `com_yahoo_elide_datastores_aggregation_example_PlayerStats_country` \n"
+                        + "            ON \n"
+                        + "            `com_yahoo_elide_datastores_aggregation_example_PlayerStats`.`country_id` = `com_yahoo_elide_datastores_aggregation_example_PlayerStats_country`.`id` \n"
+                        + "        WHERE \n"
+                        + "            (\n"
+                        + "                CASE WHEN `com_yahoo_elide_datastores_aggregation_example_PlayerStats`.`overallRating` = 'Good' THEN 1 ELSE 2 END IN (:XXX, :XXX) \n"
+                        + "                AND \n"
+                        + "                `com_yahoo_elide_datastores_aggregation_example_PlayerStats_country`.`iso_code` IN (:XXX)\n"
+                        + "            ) \n"
+                        + "        GROUP BY \n"
+                        + "            `com_yahoo_elide_datastores_aggregation_example_PlayerStats`.`overallRating`, \n"
+                        + "            CASE WHEN `com_yahoo_elide_datastores_aggregation_example_PlayerStats`.`overallRating` = 'Good' THEN 1 ELSE 2 END, \n"
+                        + "            PARSEDATETIME(FORMATDATETIME(`com_yahoo_elide_datastores_aggregation_example_PlayerStats`.`recordedDate`, 'yyyy-MM-dd'), 'yyyy-MM-dd'), \n"
+                        + "            PARSEDATETIME(FORMATDATETIME(`com_yahoo_elide_datastores_aggregation_example_PlayerStats`.`recordedDate`, 'yyyy-MM'), 'yyyy-MM') \n"
+                        + "    ) AS `com_yahoo_elide_datastores_aggregation_example_PlayerStats_XXX` \n"
+                        + "GROUP BY \n"
+                        + "    `com_yahoo_elide_datastores_aggregation_example_PlayerStats_XXX`.`overallRating_207658499`, \n"
+                        + "    `com_yahoo_elide_datastores_aggregation_example_PlayerStats_XXX`.`playerLevel_96024484`, \n"
+                        + "    PARSEDATETIME(FORMATDATETIME(`com_yahoo_elide_datastores_aggregation_example_PlayerStats_XXX`.`recordedDate_155839778`, 'yyyy-MM'), 'yyyy-MM') \n"
+                        + "HAVING \n"
+                        + "    AVG(`com_yahoo_elide_datastores_aggregation_example_PlayerStats_XXX`.`highScore`) > :XXX \n"
+                        + "ORDER BY \n"
+                        + "    `com_yahoo_elide_datastores_aggregation_example_PlayerStats_XXX`.`overallRating_207658499` DESC ";
+
+        if (useAliasForOrderByClause) {
```

Comment:
May be L-638 and L644-L646 could be 2 separate private methods in SQLUnitTest.java so that they can be used by other tests if needed.